tests/slsa_analyzer/checks/test_github_actions_vulnerability_check.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2025 - 2025, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module contains tests for the GitHub Actions vulnerabilities check."""
+
+import json
+import os
+import urllib
+from pathlib import Path
+
+import pytest
+from pytest_httpserver import HTTPServer
+
+from macaron.code_analyzer.call_graph import BaseNode, CallGraph
+from macaron.config.defaults import load_defaults
+from macaron.slsa_analyzer.checks.check_result import CheckResultType
+from macaron.slsa_analyzer.checks.github_actions_vulnerability_check import GitHubActionsVulnsCheck
+from macaron.slsa_analyzer.ci_service.base_ci_service import BaseCIService
+from macaron.slsa_analyzer.ci_service.github_actions.analyzer import build_call_graph_from_path
+from macaron.slsa_analyzer.provenance.intoto import InTotoV01Payload
+from macaron.slsa_analyzer.specs.ci_spec import CIInfo
+from macaron.slsa_analyzer.specs.inferred_provenance import InferredProvenance
+from tests.conftest import MockAnalyzeContext
+
+RESOURCE_PATH = Path(__file__).parent.joinpath("resources")
+
+
+def get_ci_info(ci_services: dict[str, BaseCIService], ci_name: str, workflow_path: str) -> CIInfo:
+    """Get CIInfo instance."""
+    ci_info = CIInfo(
+        service=ci_services[ci_name],
+        callgraph=CallGraph(BaseNode(), ""),
+        provenance_assets=[],
+        release={},
+        provenances=[],
+        build_info_results=InTotoV01Payload(statement=InferredProvenance().payload),
+    )
+    match ci_name:
```

Comment:
What is the reason for using a `match` statement here?

Reply:
I find this approach tidier than using if-else statements, and it also makes it easier to extend support for additional CI types. But I can switch to if-else if preferred, it’s not a big deal.

Reply:
See commit [a684508](https://github.com/oracle/macaron/pull/1021/commits/a6845083b70a170af22416b1f8cce556e16ca5dc).